instrumentation/tomcat-7.0/javaagent/tomcat-7.0-javaagent.gradle

```
@@ -0,0 +1,17 @@
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+muzzle {
+  pass {
+    group = "org.apache.tomcat.embed"
+    module = "tomcat-embed-core"
+    // Tomcat 10 is about servlet 5.0
+    // 7.0.4 added Request.isAsync, which is needed
+    versions = "[7.0.4, 10)"
+  }
+}
+
+dependencies {
+  library group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '9.0.40'
```

typically this is the lowest version, and then tests will be run against the latest version in nightly build using -PtestLatestDeps=true, in this case you don't quite want the latest version, so need to supply `latestDepTestLibrary` like below
```suggestion
  library group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '7.0.4'
  latestDepTestLibrary  group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '9.+'
```

---

Actually I don't need `tomcat-embed-core` for the instrumentation. I only need it for the test. For the instrumentation I need two tomcat libraries - `org.apache.tomcat:tomcat-coyote` (handling of HTTP) and `org.apache.tomcat:tomcat-catalina` (Servlet implementation). However, these two dependencies must have the same version while muzzling (because some classes moved between those Jars at some point and there are `NoClassDefFound` errors from either one or other jar). Is there a way to configure muzzle in a way that two dependencies would have the same version at a time?

---

> Is there a way to configure muzzle in a way that two dependencies would have the same version at a time?

I don't think so. The way you have it currently using the aggregating artifact `tomcat-embed-core` seems reasonable to me.